sphinxlint.py

```diff
@@ -255,11 +268,62 @@ def paragraphs(lines):
         yield paragraph_lno, "".join(paragraph)
 
 
+QUOTE_PAIRS = [
+    '»»',  # Swedish
+    '‘‚',  # Albanian/Greek/Turkish
+    '’’',  # Swedish
+    '‚‘',  # German
+    '‚’',  # Polish
+    '“„',  # Albanian/Greek/Turkish
+    '„“',  # German
+    '„”',  # Polish
+    '””',  # Swedish
+    '››',  # Swedish
+    "''",  # ASCII
+    '""',  # ASCII
+    "<>",  # ASCII
+    "()",  # ASCII
+    "[]",  # ASCII
+    "{}",  # ASCII
+]
+
+QUOTE_PAIRS_NEGATIVE_LOOKBEHIND = "(?<!" + (
+    "|".join(f"{re.escape(left)}`{re.escape(right)}" for left, right in QUOTE_PAIRS) +
+    "|" +
+    "|".join(f"{re.escape(openers[i])}`{re.escape(closers[i])}" for i in range(len(openers)))
+) + ")"
+
 role_body = rf"([^`]|\s`+|\\`|:{simplename}:`([^`]|\s`+|\\`)+`)+"
 normal_role = f":{simplename}:`{role_body}`"
 backtick_in_front_of_role = re.compile(rf"(^|\s)`:{simplename}:`{role_body}`")
 
 
+# https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-markup-recognition-rules
+default_role_re = re.compile(
+    r"""
+    (?<!\x00) # Both, inline markup start-string and end-string must not be preceded by an unescaped backslash
```

Comment:
```suggestion
    (?<!\x00) # Both inline markup start-string and end-string must not be preceded by an unescaped backslash
```

Can't you just put `\\` here instead of `\x00`?

Reply:
I use `escape2null` (from docutils) to replace escape backslash converted to nulls.

It helps a lot because ``` \\\\` ```, ``` \\\\\` ```, ``` \\\\\\` ```, ... are a tad hard to properly match with regexes.

While ``` 0\0\` ```, ``` 0\0\0` ```, ``` 0\0\0\` ``` (I replaced \x00 with 0 for human readability here) are easy to match : `` 0` `` is an escaped backtick while ``` \` ``` is just two chars.